p2p/libp2p/netMessenger.go

```diff
@@ -229,6 +231,21 @@ func createPubSub(ctxProvider *Libp2pContext, withSigning bool) (*pubsub.PubSub,
 	return ps, nil
 }
 
+func (netMes *networkMessenger) createConnectionMonitor() error {
+	reconnecter, _ := netMes.peerDiscoverer.(p2p.Reconnecter)
```

Comment:
treat if type assertion is good. You may save reconnecter directly as a private field in netWorkMessenger struct.

Reply:
If type assertions is not ok, a NilConnectionMonitor will be created. I guess it should work with or without a reconnecter.

Reply:
Yes, it works, added TODO to refactor the code